conventions/src/main/kotlin/io.opentelemetry.instrumentation.base.gradle.kts

```diff
@@ -119,3 +121,35 @@ if (testLatestDeps) {
     }
   }
 }
+
+val otelInstrumentation = extensions.create<OtelInstrumentationExtension>("otelInstrumentation")
+
+tasks {
+  val generateInstrumentationVersionFile by registering {
+    val propertiesDir = File(project.buildDir, "generated/instrumentationVersion")
+    outputs.dir(propertiesDir)
+
+    doLast {
+      val name = otelInstrumentation.name.getOrElse(project.computeInstrumentationName())
+      val version = otelInstrumentation.version.getOrElse(project.version as String)
+      val metaInfDir = File(propertiesDir, "META-INF/io/opentelemetry/instrumentation/")
+      metaInfDir.mkdirs()
+      File(metaInfDir, "$name.properties")
+        .writeText("version=$version")
+    }
+  }
+}
+
+fun Project.computeInstrumentationName(): String {
```

Comment:
I've noticed people love adding extension methods when writing Kotlin, I don't know if I get it :P But in this case especially, a normal `fun` seems like it will be the exact same code without extension. Notice that you didn't use `this.name`

Reply:
If you have a hammer... 😄 
Yeah, this definitely works better as a normal function, I suppose I just suffered from Kotlin withdrawal symptoms 🙈